internal/kojiapi/server.go

```diff
@@ -0,0 +1,162 @@
+
+	for i, ir := range request.ImageRequests {
+		arch, err := d.GetArch(ir.Architecture)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Unsupported architecture '%s' for distribution '%s'", ir.Architecture, request.Distribution), http.StatusBadRequest)
+			return
+		}
+		imageType, err := arch.GetImageType(ir.ImageType)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Unsupported image type '%s' for %s/%s", ir.ImageType, ir.Architecture, request.Distribution), http.StatusBadRequest)
+			return
+		}
+		repositories := make([]rpmmd.RepoConfig, len(ir.Repositories))
+		for j, repo := range ir.Repositories {
+			repositories[j].BaseURL = repo.Baseurl
+			repositories[j].GPGKey = repo.Gpgkey
+		}
+		packageSpecs, _ := imageType.BasePackages()
+		packages, _, err := server.rpmMetadata.Depsolve(packageSpecs, nil, repositories, d.ModulePlatformID(), arch.Name())
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Failed to depsolve base base packages for %s/%s/%s: %s", ir.ImageType, ir.Architecture, request.Distribution, err), http.StatusBadRequest)
+			return
+		}
+		buildPackageSpecs := imageType.BuildPackages()
+		buildPackages, _, err := server.rpmMetadata.Depsolve(buildPackageSpecs, nil, repositories, d.ModulePlatformID(), arch.Name())
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Failed to depsolve build packages for %s/%s/%s: %s", ir.ImageType, ir.Architecture, request.Distribution, err), http.StatusBadRequest)
+			return
+		}
+
+		manifest, err := imageType.Manifest(nil, distro.ImageOptions{Size: imageType.Size(0)}, repositories, packages, buildPackages)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Failed to get manifest for for %s/%s/%s: %s", ir.ImageType, ir.Architecture, request.Distribution, err), http.StatusBadRequest)
+			return
+		}
+
+		imageRequests[i].manifest = manifest
+	}
+
+	var ir imageRequest
+	if len(imageRequests) == 1 {
+		// NOTE: the store currently does not support multi-image composes
+		ir = imageRequests[0]
+	} else {
+		http.Error(w, fmt.Sprintf("Only single-image composes are currently supported"), http.StatusBadRequest)
+		return
+	}
+	id, err := server.workers.Enqueue(ir.manifest, nil)
+	if err != nil {
+		// This is a programming errror.
+		panic(err)
+	}
+
+	var response ComposeResponse
+	response.Id = id.String()
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	w.WriteHeader(http.StatusCreated)
+	err = json.NewEncoder(w).Encode(response)
+	if err != nil {
+		panic("Failed to write response")
+	}
+}
+
+// GetComposeId handles a /compose/{id} GET request
+func (server *Server) GetComposeId(w http.ResponseWriter, r *http.Request) {
+	id, err := uuid.Parse(r.Context().Value("id").(string))
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Invalid format for parameter id: %s", err), http.StatusBadRequest)
+		return
+	}
+
+	status, err := server.workers.JobStatus(id)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Job %s not found: %s", id.String(), err), http.StatusBadRequest)
```

Comment:
This should probably be `http.StatusNotFound`

Reply:
Actually `server.workers.JobStatus` returns 400 when it can't find an ID, that should probably be fixed?